modules/bullet/bullet_physics_server.cpp

```diff
@@ -518,6 +525,13 @@ void BulletPhysicsServer::body_set_shape_transform(RID p_body, int p_shape_idx,
 	body->set_shape_transform(p_shape_idx, p_transform);
 }
 
+void BulletPhysicsServer::body_set_shape_margin(RID p_body, int p_shape_idx, real_t p_margin) {
```

Comment:
You have to delete this function and instead create new one called `shape_set_margin` that accept shape RID

Reply:
How can I modify the `ShapeData.margin` inside the collision object used by `CollisionObject::shape_owner_add_shape` ?

Reply:
You don't want to add that parameter there, because the margin is a properties of the shape. So you need to add it to the shape and only the shape must be aware of it

Reply:
It seems that currently the only way to set the margin is to recreate the shape and pass indirectly to function create_bt_shape the margin.

Bullet is supporting modifying the margin of existing shapes (with an extra AABB re-computation), do you want me to implement a way to do it or continue with re-creating the shape at each margin modification ?

Reply:
I don't like recreate the shape on each margin modification, but to keep thing simple for now is better pass margin to the shape and then leave the event recreate the shape with the new margin.

At least the shape modification doesn't occurs so often so it's fine for now